tools/src/dwarfs_main.cpp

```diff
@@ -1588,6 +1603,39 @@ int dwarfs_main(int argc, sys_char** argv, iolayer const& iol) {
 
   fuse_opt_parse(&args, &userdata.opts, dwarfs_opts.data(), option_hdl);
 
+  // automount to a new/existing dir
+  // TODO: check already mounted. refactor extract function.
+  if (userdata.opts.is_automount) {
+    if (userdata.opts.seen_mountpoint) {
+      iol.err << "error: cannot combine <mountpoint> with --automount" << "\n";
+      usage(iol.out, userdata.progname);
+      return 1;
+    }
+    auto fspath = std::filesystem::path(userdata.opts.fsimage->data());
+    // First try without file extension, then add "_mount". Otherwise, fail.
+    // This way, user gets:
+    //   "fs.dwarfs"  -> "fs/"
+    //   "fs"         -> "fs_mount/"
+    auto mountpath = fspath.parent_path() / fspath.stem();
+    if (!is_good_mount_path(mountpath)) {
+      mountpath += "_mount";
+      if (!is_good_mount_path(mountpath)){
+        iol.err << "error: cannot find a suitable mount dir" << "\n";
+        return 1;
+      }
+    }
+    if (!std::filesystem::exists(mountpath)) {
+      if (!std::filesystem::create_directory(mountpath)) {
+        iol.err << "error: unable to create mount dir" << "\n";
+        return 1;
+      }
+    }
```

Comment:
There's a potential race here between `exists` and `create_directory`. Also, it's good to see *why* it was unable to create the directory.
```suggestion
    if (!std::filesystem::exists(mountpath)) {
      std::error_code ec;
      if (!std::filesystem::create_directory(mountpath, ec)) {
        if (ec != std::errc::file_exists) {
          iol.err << "error: unable to create mountpoint directory: " << ec.message() << "\n";
          return 1;
        }
      }
    }
```